mlflow/telemetry/utils.py

```diff
@@ -14,8 +15,31 @@ def is_telemetry_disabled() -> bool:
     )
 
 
-# TODO: add whitelist for APIs that's only invoked by MLflow, e.g. MlflowV2SpanExporter.export
-def invoked_from_internal_api() -> bool:
+def _get_whitelist() -> dict[str, list[str]]:
+    """
+    Whitelist for APIs that are only invoked by MLflow but should be tracked.
+    """
+    whitelist = defaultdict(list)
+    try:
+        from mlflow.pyfunc.utils.data_validation import _infer_schema_from_list_type_hint
+
+        whitelist[_infer_schema_from_list_type_hint.__module__].append(
+            _infer_schema_from_list_type_hint.__qualname__
+        )
+    except ImportError:
+        pass
+
+    return whitelist
+
+
+def invoked_from_internal_api(func) -> bool:
```

Comment:
nit: we might want to change the naming since the APIs that are included in the whitelist is technically "invoked from internal api"